server/fleet/microsoft_mdm.go

```diff
@@ -199,14 +199,49 @@ func (req *SoapRequest) IsValidRequestSecurityTokenMsg() error {
 		return errors.New("invalid requestsecuritytoken message: BinarySecurityToken.ValueType")
 	}
 
-	if len(req.Body.RequestSecurityToken.BinarySecurityToken.EncodingType) == 0 {
-		return errors.New("invalid requestsecuritytoken message: BinarySecurityToken.EncodingType")
+	if req.Body.RequestSecurityToken.BinarySecurityToken.ValueType != mdm.EnrollReqTypePKCS10 &&
+		req.Body.RequestSecurityToken.BinarySecurityToken.ValueType != mdm.EnrollReqTypePKCS7 {
+		return errors.New("invalid requestsecuritytoken message: BinarySecurityToken.EncodingType  not supported")
 	}
 
 	if len(req.Body.RequestSecurityToken.BinarySecurityToken.Content) == 0 {
 		return errors.New("invalid requestsecuritytoken message: BinarySecurityToken.Content")
 	}
 
+	if len(req.Body.RequestSecurityToken.AdditionalContext.ContextItems) == 0 {
+		return errors.New("invalid requestsecuritytoken message: AdditionalContext.ContextItems")
+	}
+
+	reqVersion, err := req.Body.RequestSecurityToken.GetContextItem(mdm.ReqSecTokenContextItemRequestVersion)
+	if err != nil || (reqVersion != mdm.MaxEnrollmentVersion && reqVersion != mdm.MinEnrollmentVersion) {
```

Comment:
Should the version be compared as >= min and <= max? The constants are min and max but they are really the only two exact versions allowed as it stands (nothing in between).

Reply:
good point, I've renamed the variable names to avoid confusion